cmd/virt-chroot/tap-device-maker.go

```diff
@@ -16,18 +16,14 @@ func createTapDevice(name string, owner uint, group uint, queueNumber int, mtu i
 		LinkAttrs:  netlink.LinkAttrs{Name: name},
 		Mode:       unix.IFF_TAP,
 		NonPersist: false,
-		Queues:     queueNumber,
 		Owner:      uint32(owner),
 		Group:      uint32(group),
 	}
 
-	// when netlink receives a request for a tap device with 1 queue, it uses
-	// the MULTI_QUEUE flag, which differs from libvirt; as such, we need to
-	// manually request the single queue flags, enabling libvirt to consume
-	// the tap device.
-	// See https://github.com/vishvananda/netlink/issues/574
 	if queueNumber == 1 {
 		tapDevice.Flags = netlink.TUNTAP_DEFAULTS
+	} else {
+		tapDevice.Flags = netlink.TUNTAP_MULTI_QUEUE_DEFAULTS
```

Comment:
can't we set this one all the time?

Reply:
AFAIK, libvirt uses this flag only if the number of queues (defined on the interface) is greater than one [1].

[1] https://github.com/libvirt/libvirt/blob/1899d7df34576e414c49e4d91c68e6e2348f37f2/src/util/virnetdevtap.c#L218-L219